### PR TITLE
Interstellar update

### PR DESCRIPTION
**Pull request by ashfurrow**
Continues from #1428. We need to use the new `Observable` class instead of `Signal`, since it allows us to unsubscribe from updates (this is very important for performance reasons). 

One important note about the change: `Observable` removes the concept of errors that `Signal` had. For signals that require errors, we need to use `Result` as the `Observable`'s value. There are extensions for `Observable`'s of `Result`s, which is pretty cool. I like this personally; while if it requires us to be slightly more verbose, it also encodes the errorability of an `Observable` into the type system.

We need to use my fork until https://github.com/JensRavens/Interstellar/pull/39 is merged. Also, this led to discovery of a Swift compiler bug ( https://bugs.swift.org/browse/SR-1313 ), even though it's a duplicate.

Still to be done:

- [x] Unsubscribe from `Observable`s where required.

### CODE REVIEW

**Comment by ashfurrow** [comment]
I'm not loving the clunkiness around unsubscribing from observables, I've opened https://github.com/JensRavens/Interstellar/issues/40 for discussion, but this PR can be reviewed independently.

**Comment by ArtsyMobile** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%">1 Warning</th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:warning:</td>
      <td>Needs testing on a Phone if change is non-trivial</td>
    </tr>
    
  </tbody>
</table>
  

  


<p align="right" data-meta="generated_by_danger" data-base-commit="" data-head-commit="" >
  Generated by :no_entry_sign: <a href="https://github.com/danger/danger/">danger</a>
</p>

File: Artsy/View_Controllers/Auction/AuctionNetworkModel.swift [hunk 1]
```diff
@@ -2,8 +2,8 @@ import Foundation
 import Interstellar
 
 protocol AuctionNetworkModelType {
-    func fetch() -> Signal<SaleViewModel>
-    func fetchRegistrationStatus() -> Signal<ArtsyAPISaleRegistrationStatus>
+    func fetch() -> Observable<Result<SaleViewModel>>
```
Comment (mennenia) [inline Artsy/View_Controllers/Auction/AuctionNetworkModel.swift]: so a `Observable<Result<x>>`` is an observable that can error, right?

I like that it's explicitly stated like that!

Comment (ashfurrow) [inline Artsy/View_Controllers/Auction/AuctionNetworkModel.swift]: Exactly! I was a little skeptical at first, but I really like it.

File: Artsy/View_Controllers/Live_Auctions/LiveAuctionLotViewController.swift [hunk 2]
```diff
@@ -31,6 +35,18 @@ class LiveAuctionLotViewController: UIViewController {
         return nil
     }
 
+    deinit {
+        if let currentLotObserver = currentLotObserver {
+            currentLotSignal.unsubscribe(currentLotObserver)
```
Comment (mennenia) [inline Artsy/View_Controllers/Live_Auctions/LiveAuctionLotViewController.swift]: yay for unsubscribing!

**Comment by mennenia** [comment]
Looking all dandy! 🎉

**Pull request merged**